packages/flutter_tools/lib/src/device.dart

```diff
@@ -283,8 +283,10 @@ abstract class PollingDeviceDiscovery extends DeviceDiscovery {
   }
 
   @override
-  Future<List<Device>> get devices async {
-    _items ??= ItemListNotifier<Device>.from(await pollingGetDevices());
+  Future<List<Device>> devices({ Duration timeout, bool useCache = true }) async {
+    if (_items == null || !useCache) {
+      _items = ItemListNotifier<Device>.from(await pollingGetDevices(timeout: timeout));
```

Comment:
I needed to add a mechanism to not used the `_items` cache because in my next commit I'm going to be doing a quick default timeout, then longer when a particular ID is specified but not found.  But since you guys pointed out how annoying adding default params is in g3 I'm going to do both in the same PR in case they get split over rolls.